src/main/java/com/lb/application/TripCoordinator.java

```diff
@@ -0,0 +1,135 @@
+                        Send an email to the email provided in %s, using the requestId provided in %s and the content from %s and %s. The content has flights and accommodations
+                        Add in the email a recommendation with the best value combination flight (outbound and return) and accommodation
+                        parse the whole content as HTML before sending
+                        """,
+                    question, question, responseFlights, responseAccommodations))
+            .tools(new EmailAPITool())
+            .call()
+            .content();
+    log.debug(String.format("responseMail %s", responseMail));
+    return CompletableFuture.completedFuture(
+        "We are processing your request. We'll send you the response to your email in a minute.");
+  }
+
+  public CompletionStage<String> bookTrip(BookingTripRequest bookingTripRequest) {
+    return componentClient
+        .forEventSourcedEntity(bookingTripRequest.flightRef())
+        .method(FlightBookingEntity::book)
+        .invokeAsync()
+        .thenApply(__ -> String.format("Flight %s booked", bookingTripRequest.flightRef()));
+  }
+
+  public static String extractJson(String response) {
+    Pattern pattern = Pattern.compile("\\[.*?]", Pattern.DOTALL);
+    Matcher matcher = pattern.matcher(response);
+    if (matcher.find()) {
+      return matcher.group(0);
+    }
+    String message = "Could not extract json from response";
+    log.error("{}: {}", message, response);
+    throw new IllegalArgumentException(message);
+  }
+
+  private void storeAccommodations(String chatResponseAccomodations) {
+    log.debug("parsing accommodations: {}", chatResponseAccomodations);
+    String onlyAccommodations = extractJson(chatResponseAccomodations);
+    List<AccommodationAPIResponse> accommodationAPIResponses =
+        AccommodationAPIResponse.extract(new ByteArrayInputStream(onlyAccommodations.getBytes()));
+    log.debug("storing accommodations: {}", accommodationAPIResponses);
+    // load accommodations into entities
+    accommodationAPIResponses.forEach(
+        accommodation -> {
+          componentClient
+              .forEventSourcedEntity(accommodation.id())
+              .method(AccommodationBookingEntity::create);
+        });
+  }
+
+  private void storeFlights(String chatResponseFlights) {
+    log.debug("parsing flights: {}", chatResponseFlights);
+    String onlyFlights = extractJson(chatResponseFlights);
+    List<FlightAPIResponse> flightAPIResponses =
+        FlightAPIResponse.extract(new ByteArrayInputStream(onlyFlights.getBytes()));
+    log.debug("storing flights: {}", flightAPIResponses);
+    // load flights into entities
+    flightAPIResponses.forEach(
+        flight -> {
+          componentClient.forEventSourcedEntity(flight.id()).method(FlightBookingEntity::create);
```

Comment:
This is not doing anything, it's not calling invoke. Will be easier with non-async.

Reply:
added the invoke